### ISSUE TITLE
website.configurator_s_media_list doesn't exist

### ISSUE THREAD

**Issue by lflfm** [odoo/design-themes#754]
The themes below try to extend `website.configurator_s_media_list` template, which doesn't exist.

- theme_notes
- theme_odoo_experts
- theme_paptic

Checked in the 17.0 branch.

**Comment by qsm-odoo** [comment]
Hello @lflfm 

The template does exist, it is one of the new many automatically generated views in 17.0. This one in particular (`website.configurator_s_media_list`) will be the one that is used when a page created with the website configurator uses the s_media_list block. By default it is equal to the s_media_list block but if a theme extends the template you mentioned, it means that the pages created with the website configurator and with this theme will use the extension but the s_media_list block that is drag&droppable via the editor will not.

For instance, I create this website with theme_notes, you see that the homepage uses that block and it has a shape on it:
![image](https://github.com/odoo/design-themes/assets/10338094/ac93d8e7-5112-49f7-9a58-0ae9897787d9)

But if I drop the same block in one page, I don't have the shape because it is part of the `website.configurator_s_media_list` extension:
![image](https://github.com/odoo/design-themes/assets/10338094/3aa833ff-79b9-44f9-95a9-e70a813591fa)

And you can see the related view does exist:
![image](https://github.com/odoo/design-themes/assets/10338094/689ad515-2dc5-401c-89e6-b8a0a9ac363c)

If you have an issue with this system on your side, please tell us how we can reproduce it, we'll have a look :+1: 

Thank you!

**Comment by lflfm** [comment]
Hi @qsm-odoo! Many thanks for the quick response and example!

Yes, we get issues with this when updating modules or switching themes, since the XML file is trying to reference the templates that don't _yet_ exist - I guess we can make an analogy here basically as if we try to use a template that is defined later in the same file - except in this case it's not in any file and will be defined later.

The website is then "stuck" with the theme as we get these validation errors trying to switch.

![Screenshot_20240109_140200](https://github.com/odoo/design-themes/assets/26166377/84c25d47-d721-4627-a441-05de6289e552)

I also run some Odoo instances based on the bitnami container image and it runs `--update all` on init, which error out on this error.

I guess the solution would be to make these templates also dynamic...?

**Comment by qsm-odoo** [comment]
Hello @lflfm 

> Yes, we get issues with this when updating modules or switching themes, since the XML file is trying to reference the templates that don't yet exist

This is not supposed to be possible since to be able to update a 17.0 theme, the database should have been upgraded to 17.0 first. I tested the flow again: if I create a 16.x database, install the theme note, upgrade to 17.0 and then try to re-update the theme, it works: when upgrading to 17.0, the `website.configurator_XXX` views were created and can thus be referenced by themes.

> I also run some Odoo instances based on the bitnami container image and it runs --update all on init, which error out on this error.

I don't know much about this but normally, any `-u website` should create the `website.configurator_XXX` views automatically.

Can you confirm in which exact case your get the error? Are you trying to upgrade a 16.0 database or do you already have a working 17.0 which has this problem?

**Comment by lflfm** [comment]
Hi  @qsm-odoo!
I'm trying this on newly created Odoo 17 environments.
The issue seems to be more directly related to the `--update all` option:

1. install Odoo base on new database, start Odoo (community)
2. log in, install website module
3. follow first site tutorial, pick any theme
4. shut down the server
5. start server with update all, eg. `odoo-bin --update all`
6. if no error, shut down and start again with update all - don't open the website editor between the two server starts, the error seems to be more consistent this way

I don't know how the error became "permanent" on that environment I had where I took that screenshot on the error on the frontend, I didn't manage to replicate it and already deleted the environment.

I think the website editor might "fix" the problem before it occurs between two updates (the problem is that, in a container environment, restarts can be quite frequent and, because these containers may run the update at start, then the entire environment might be "disabled" by this problem).

This might not be related, but based on what you said I thought it might be worth mentioning:
I did notice that, when I update the website module alone in the apps app, I get these messages in the log, which do not show up anywhere when running the `--update all` option - I'm thinking they would have shown up after validation of the theme files, if those hadn't failed during this process...
```
2024-01-09 17:45:15,156 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 10 primary snippet templates for 'theme_anelusia' 
2024-01-09 17:45:15,236 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 2 primary snippet templates for 'theme_avantgarde' 
2024-01-09 17:45:15,265 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 2 primary snippet templates for 'theme_aviato' 
2024-01-09 17:45:15,301 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 2 primary snippet templates for 'theme_bewise' 
2024-01-09 17:45:15,334 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 6 primary snippet templates for 'theme_bistro' 
2024-01-09 17:45:15,429 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 8 primary snippet templates for 'theme_bookstore' 
2024-01-09 17:45:15,467 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 2 primary snippet templates for 'theme_buzzy' 
2024-01-09 17:45:15,530 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 11 primary snippet templates for 'theme_default' 
2024-01-09 17:45:15,573 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 1 primary snippet templates for 'theme_graphene' 
2024-01-09 17:45:15,682 5850 INFO db_odoo_community odoo.addons.website.models.ir_module_module: Generated 2 primary snippet templates for 'theme_nano' 
```

**Comment by qsm-odoo** [comment]
I actually tried with `-u all` but not on a 17.0 database I think, I tested an upgrade from 16.4 to 17.0, and with our upgrade repo. I also tested in 17.0 only but in that case probably with `-u website`, not `-u all`. Might be relevant.

@bso-odoo Could you have a look at if you can reproduce the issue on your side? You may have more ideas about this than me anyway :+1: Thanks :+1:

**Comment by bso-odoo** [comment]
@qsm-odoo I will try to reproduce it, but to fix it I guess I will have to apply what I prepared for master in <https://github.com/odoo/design-themes/pull/755>.

**Comment by bso-odoo** [comment]
@qsm-odoo I only manage to reproduce a similar error by using a version before https://github.com/odoo/odoo/commit/2b2122a3c69ff356a43221d77fafdede88ad72ec, by doing this:
```sh
odoo-bin -i website # generates templates for all themes
odoo-bin -u all # the configurator_* template were deleted
odoo-bin -i theme_paptic # => fail because template were missing
```

To be on the safe side though, I think that I should rebase https://github.com/odoo/design-themes/pull/755 (and https://github.com/odoo/odoo/pull/148443) onto 17.0 because it is a much more reasonable way to generate those templates at the right time.

**Comment by rdeodoo** [comment]
Let's mention this issue on top of task-3670496 and I'll have a look once it's rebased :+1:

**Comment by bso-odoo** [comment]
@lflfm FYI task-3670496 has been merged. It should solve the problem addressed by this PR.

Merged into 17.0:
- https://github.com/odoo/odoo/commit/255eb4c99c04da47b03320f8a760e09f23f9454e
- https://github.com/odoo/design-themes/commit/599984eea17c0683ff266476f29a419fae4d682e
- https://github.com/odoo/design-themes/commit/4cdd1db20c021ebd369b46c739591def9732eec5

Additionally merged into saas-17.1:
- https://github.com/odoo/odoo/commit/9ae5e2bb25ea164264aeab7e45f53472a7d6aa5a
- https://github.com/odoo/design-themes/commit/b8aae07df41b44aa78cf39ff6104136556199130
